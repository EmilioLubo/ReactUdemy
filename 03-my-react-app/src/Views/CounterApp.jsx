import { useState } from 'react';
import PropTypes from 'prop-types';
import { Button ,ButtonGroup } from 'react-bootstrap';

export const CounterApp = ({value}) =>{
    const [counter, setCounter] = useState(value);

    const handleAdd = () => setCounter(counter + 1);
    const handleSubstract = () => setCounter(counter -1);
    const handleReset = () => setCounter(value);

    return(
        <header className='d-flex flex-column align-items-center'>
                <h1 className='text-center'>CounterApp</h1>
                <h2 className='text-center'>{ counter }</h2>
                <ButtonGroup>
                    <Button variant="warning" className="me-1" onClick={ handleSubstract }>-1</Button>
                    <Button variant="danger" className="me-1" onClick={ handleReset }>Reset</Button>
                    <Button variant="success" onClick={ handleAdd }>+1</Button>
                </ButtonGroup>
        </header>
    );
}
CounterApp.propTypes = {
    value: PropTypes.number
}