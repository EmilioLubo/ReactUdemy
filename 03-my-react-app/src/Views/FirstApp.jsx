import PropTypes from 'prop-types';
const user = {
    uName: "5",
    anoNac: 1987,
    message: "Hola"
}

const saludar = (saludo, aQuien) => {
        if(aQuien.length < 1){
            return `${saludo} extraño/a!` ;
        } else{
            return `${saludo} ${aQuien}!`;
        }
    }

export const FirstApp = ({title, subtitulo}) =>   {
    return(
        <>
            <h1 data-testid="test-title"> {title} </h1>
            <h2>{saludar(user.message, user.uName)}</h2>
            {<code>{JSON.stringify({user})}</code>}
            <p>{subtitulo}</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
    subtitulo: 'hermoso día'
}