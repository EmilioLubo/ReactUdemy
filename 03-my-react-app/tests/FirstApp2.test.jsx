import { getAllByText, render, screen } from '@testing-library/react';
import { FirstApp } from '../src/Views/FirstApp';

describe('Pruebas en <FirstApp/>', () => {

    const title = "test title";
    const subtitulo = "elsub";

    test('debe de hacer match con el snapshot', () => {
        const {container} = render(<FirstApp title = {title} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje "test title"', () => {
        render(<FirstApp title = {title} />);
        expect(screen.getByText(title)).toBeTruthy();
        //screen.debug();
    });

    test('debe de mostrar el tÃ­tulo en un h1', () => {
        render(<FirstApp title = {title} />);
        expect(screen.getByRole('heading', {level: 1}))
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        render(
            <FirstApp 
                title = {title} 
                subtitulo = {subtitulo} 
            />);
        expect(screen.getAllByText(subtitulo).length).toBe(1);
    });
   
});