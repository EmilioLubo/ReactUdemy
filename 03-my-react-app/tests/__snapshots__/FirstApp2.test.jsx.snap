// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    test title
     
  </h1>
  <h2>
    Hola 5!
  </h2>
  <code>
    {"user":{"uName":"5","anoNac":1987,"message":"Hola"}}
  </code>
  <p>
    hermoso d√≠a
  </p>
</div>
`;
